pkg/api/handlers.go

```diff
@@ -207,25 +207,39 @@ func NewQuerierHandler(
 	middlewares := middleware.Merge(inst, cacheGenHeaderMiddleware)
 	router.Use(middlewares.Wrap)
 
-	promRouter := route.New().WithPrefix(cfg.ServerPrefix + cfg.PrometheusHTTPPrefix + "/api/v1")
+	// Define the prefixes for all routes
+	prefix := cfg.ServerPrefix + cfg.PrometheusHTTPPrefix
+	legacyPrefix := cfg.ServerPrefix + cfg.LegacyHTTPPrefix
+
+	promRouter := route.New().WithPrefix(prefix + "/api/v1")
 	api.Register(promRouter)
 
-	legacyPromRouter := route.New().WithPrefix(cfg.ServerPrefix + cfg.LegacyHTTPPrefix + "/api/v1")
+	legacyPromRouter := route.New().WithPrefix(legacyPrefix + "/api/v1")
 	api.Register(legacyPromRouter)
 
-	//TODO(gotjosh): This custom handler is temporary until we're able to vendor the changes in:
+	// TODO(gotjosh): This custom handler is temporary until we're able to vendor the changes in:
 	// https://github.com/prometheus/prometheus/pull/7125/files
-	router.Path(cfg.PrometheusHTTPPrefix + "/api/v1/metadata").Handler(querier.MetadataHandler(distributor))
-	router.Path(cfg.PrometheusHTTPPrefix + "/api/v1/read").Handler(querier.RemoteReadHandler(queryable))
-	// A prefix is fine because external routes will be registered explicitly
-	router.PathPrefix(cfg.PrometheusHTTPPrefix + "/api/v1/").Handler(promRouter)
-
-	//TODO(gotjosh): This custom handler is temporary until we're able to vendor the changes in:
+	router.Path(prefix + "/api/v1/metadata").Handler(querier.MetadataHandler(distributor))
+	router.Path(prefix + "/api/v1/read").Handler(querier.RemoteReadHandler(queryable))
+	router.Path(prefix + "/api/v1/read").Methods("POST").Handler(promRouter)
+	router.Path(prefix+"/api/v1/query").Methods("GET", "POST").Handler(promRouter)
```

Comment:
Nit: could we have spaces around each `+` to let the eye quickly follow the pattern?

Reply:
That bothered me as well. I tried to fix it, but the`gofmt` lint complains for some reason.

Reply:
:cry: one of the first times I am not ok with gofmt. Thanks for trying!

Reply:
Same haha